pkg/framework/test/apiserver.go

```diff
@@ -33,24 +31,22 @@ type certDirManager interface {
 //go:generate counterfeiter . certDirManager
 
 // NewAPIServer creates a new APIServer Fixture Process
-func NewAPIServer(pathToAPIServer, apiServerURL, etcdURL string) *APIServer {
+func NewAPIServer(pathToAPIServer string) *APIServer {
 	starter := func(command *exec.Cmd, out, err io.Writer) (SimpleSession, error) {
 		return gexec.Start(command, out, err)
 	}
 
 	apiserver := &APIServer{
 		Path:           pathToAPIServer,
-		EtcdURL:        etcdURL,
 		ProcessStarter: starter,
 		CertDirManager: NewTempDirManager(),
-		APIServerURL:   apiServerURL,
 	}
 
 	return apiserver
 }
 
 // Start starts the apiserver, waits for it to come up, and returns an error, if occoured.
-func (s *APIServer) Start() error {
+func (s *APIServer) Start(config map[string]string) error {
```

Comment:
I'm not a very big fan of the map[string]string for configuration. I understand that we can't pass in a specific structure because `Start` is part of an interface, but then can we pass in the options at creation time? That's a big tell that something is going wrong here I think.

Reply:
I understand that `map[string]string` is not really good. I went with that, because it was an easy first step. Anyway, I was thinking about some sort of `FixtureConfig` interface, that one can pass to `Start` and which is implemented by an `EtcdConfig` and an `APIServerConfig`.